torch/distributed/elastic/rendezvous/dynamic_rendezvous.py

```diff
@@ -1188,25 +1189,31 @@ def next_rendezvous(self) -> RendezvousInfo:
                 bootstrap_store_info,
             )
 
+        # This will only be hit when TCPStore sharing is enabled.
         if self._bootstrap_store_info is None:
+            # To avoid race in get_free_port because we release the port after the call,
+            # we want to create a TCPStore server soon afterwards.
+            if rank == 0:
+                # Since we create TCPStore with `multi_tenant = True`, we won't create
+                # a new TCPStore service if there is already one running.
+                if isinstance(self._store, dist.TCPStore):
+                    self._shared_tcp_store_server = self._store
```

Comment:
I don't think we need this branch anymore

Can just nuke it since we always want to start a clean store